src/Waives.Pipelines/DocumentProcessor.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Waives.Pipelines
+{
+    internal class DocumentProcessor : IDocumentProcessor
+    {
+        private readonly Func<Document, Task<WaivesDocument>> _docCreator;
```

Comment:
✋ Throughout the SDK, we've been careful not to abbreviate "Document" to "Doc". Please could you rename the variables, parameters, etc., using the abbreviation in this PR to be consistent with the rest of the library?